wrappers/wrapper.rb

```diff
@@ -572,6 +573,88 @@ def detect_unfeasible_services(vrp)
       unfeasible
     end
 
+    def sequence_relation_with_no_reachable_vehicle?(unfeasible, vrp, relation)
+      return false unless Models::Relation::POSITION_TYPES.include?(relation.type)
+
+      vrp.vehicles.each{ |v|
+        unreachable_service_id = last_service_reachable_sequence(vrp, v, relation.linked_services)
+        next unless unreachable_service_id
+
+        service_met = false
+        relation.linked_services.each{ |service|
+          service_met = true if service.id == unreachable_service_id
+          next unless service_met
+
+          service.vehicle_compatibility[v.id] = false
+        }
+      }
+
+      previous_removed = false
+      to_delete_services = []
+      relation.linked_services.each{ |service|
+        previous_removed ||= vrp.vehicles.none?{ |vehicle|
+          service.vehicle_compatibility[vehicle.id]
+        }
+
+        if previous_removed
+          if Models::Relation::ALL_OR_NONE_RELATIONS.include?(relation.type)
+            to_delete_services = relation.linked_services
+            break false
+          else
+            to_delete_services << service
+          end
+        end
+      }
+
+      relation.linked_services -= to_delete_services
```

Comment:
I am not sure if we need to do this. The linked_service_ids/linked_services are parsed in ortools.rb so that only the ids/services that are present in the problem are passed to the solver.

Reply:
If services are removed from the vrp and put in the unassigned list, if we perform as vrp.as_json, the relation will contain non existing service_ids which will generate error within the Vrp.create